File: solution/2807. Insert Greatest Common Divisors in Linked List/2807.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* insertGreatestCommonDivisors(ListNode* head) {
        auto p = head;
        while (p && p->next) {
            p->next = new ListNode(gcd(p->val, p->next->val), p->next);
            p = p->next->next;
        }
        return head;
    }
};

File: solution/2807. Insert Greatest Common Divisors in Linked List/2807.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def insertGreatestCommonDivisors(self, head: Optional[ListNode]) -> Optional[ListNode]:
        p = head;
        while p and p.next:
            # p.next, p = ListNode(gcd(p.val, p.next.val), p.next), p.next
            p.next = ListNode(gcd(p.val, p.next.val), p.next)
            p = p.next.next
        return head
    
